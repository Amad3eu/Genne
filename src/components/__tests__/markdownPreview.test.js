import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MarkdownPreview, {
  GithubProfileTrophyPreview,
  GitHubStatsPreview,
  SkillsPreview,
  SocialPreview,
  SubTitlePreview,
  TitlePreview,
  TopLanguagesPreview,
  TwitterBadgePreview,
  VisitorsBadgePreview,
  WorkPreview,
  SectionTitle,
  DisplayWork,
  DisplaySocial,
} from '../markdownPreview';

configure({ adapter: new Adapter() });

const DEFAULT_PREFIX = {
  title: "Hi 👋, I'm",
  currentWork: '🔭 I’m currently working on',
  currentLearn: '🌱 I’m currently learning',
  collaborateOn: '👯 I’m looking to collaborate on',
  helpWith: '🤝 I’m looking for help with',
  ama: '💬 Ask me about',
  contact: '📫 How to reach me',
  resume: '📄 Know about my experiences',
  funFact: '⚡ Fun fact',
  portfolio: '👨‍💻 All of my projects are available at',
  blog: '📝 I regularly write articles on',
};

const DEFAULT_DATA = {
  title: 'dummy',
  subtitle: 'A passionate frontend developer from India',
  currentWork: 'readme-generator',
  currentLearn: '',
  collaborateOn: '',
  helpWith: '',
  ama: '',
  contact: '',
  funFact: '',
  twitterBadge: false,
  visitorsBadge: false,
  badgeStyle: 'flat',
  badgeColor: '0e75b6',
  badgeLabel: 'Profile views',
  githubProfileTrophy: false,
  githubStats: false,
  githubStatsOptions: {
    theme: '',
    titleColor: '',
    textColor: '',
    bgColor: '',
    hideBorder: false,
    cacheSeconds: null,
    locale: 'en',
  },
  topLanguages: false,
  topLanguagesOptions: {
    theme: '',
    titleColor: '',
    textColor: '',
    bgColor: '',
    hideBorder: false,
    cacheSeconds: null,
    locale: 'en',
  },
  devDynamicBlogs: false,
  mediumDynamicBlogs: false,
  rssDynamicBlogs: false,
};

const DEFAULT_LINK = {
  currentWork: 'https://dummy.com',
  collaborateOn: '',
  helpWith: '',
  portfolio: '',
  blog: '',
  resume: '',
};

const DEFAULT_SOCIAL = {
  github: '',
  dev: '',
  linkedin: '',
  codepen: 'dummy',
  stackoverflow: '',
  kaggle: '',
  codesandbox: '',
  fb: '',
  instagram: '',
  twitter: '',
  dribbble: '',
  behance: '',
  medium: '',
  youtube: '',
  codechef: '',
  hackerrank: '',
  codeforces: '',
  leetcode: '',
  topcoder: '',
  hackerearth: '',
  geeks_for_geeks: '',
  discord: '',
  rssurl: '',
};

const DUMMY_SKILLS = {
  skills: {
    unity: true,
    android: false,
    angularjs: false,
    apachecordova: false,
  },
};

describe('Markdown Preview', () => {
  it('renders correctly', () => {
    let prefix = DEFAULT_PREFIX;
    let data = DEFAULT_DATA;
    let link = DEFAULT_LINK;
    let social = DEFAULT_SOCIAL;
    let skills = {};
    const tree = shallow(<MarkdownPreview prefix={prefix} data={data} link={link} social={social} skills={skills} />);

    expect(tree).toMatchSnapshot();
  });
});
describe('Title Preview', () => {
  it('renders correctly', () => {
    let prefix = DEFAULT_PREFIX;
    let data = DEFAULT_DATA;
    const tree = shallow(<TitlePreview prefix={prefix.title} title={data.title} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no prefix', () => {
    let prefix = DEFAULT_PREFIX;
    const tree = shallow(<TitlePreview prefix={prefix.title} title={''} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no title', () => {
    let data = DEFAULT_DATA;
    const tree = shallow(<TitlePreview title={data.title} prefix={''} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no title and prefix', () => {
    const tree = shallow(<TitlePreview />);

    expect(tree).toMatchSnapshot();
  });
});
describe('SubTitle Preview', () => {
  it('renders correctly', () => {
    let data = DEFAULT_DATA;
    const tree = shallow(<SubTitlePreview subtitle={data.subtitle} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no subtitle', () => {
    const tree = shallow(<SubTitlePreview subtitle={''} />);

    expect(tree).toMatchSnapshot();
  });
});
describe('SectionTitle Preview', () => {
  it('renders correctly', () => {
    const tree = shallow(<SectionTitle visible={true} label={'dummy'} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no label', () => {
    const tree = shallow(<SectionTitle visible={true} label={''} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with visible false', () => {
    const tree = shallow(<SectionTitle visible={false} label={'dummy'} />);

    expect(tree).toMatchSnapshot();
  });
});
describe('DisplayWork Preview', () => {
  it('renders correctly', () => {
    let prefix = DEFAULT_PREFIX;
    let data = DEFAULT_DATA;
    let link = DEFAULT_LINK;
    const tree = shallow(<DisplayWork prefix={prefix} project={data.currentWork} link={link.currentWork} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no prefix, link and project', () => {
    const tree = shallow(<DisplayWork prefix={undefined} project={undefined} link={undefined} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no prefix', () => {
    let data = DEFAULT_DATA;
    let link = DEFAULT_LINK;
    const tree = shallow(<DisplayWork prefix={undefined} project={data.currentWork} link={link.currentWork} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no project', () => {
    let prefix = DEFAULT_PREFIX;
    let link = DEFAULT_LINK;
    const tree = shallow(<DisplayWork prefix={prefix} project={undefined} link={link.currentWork} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no link', () => {
    let prefix = DEFAULT_PREFIX;
    let data = DEFAULT_DATA;
    const tree = shallow(<DisplayWork prefix={prefix} project={data.currentWork} link={undefined} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no prefix and link', () => {
    let data = DEFAULT_DATA;
    const tree = shallow(<DisplayWork project={data.currentWork} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no project and link', () => {
    let prefix = DEFAULT_PREFIX;
    const tree = shallow(<DisplayWork prefix={prefix} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no project and prefix', () => {
    let link = DEFAULT_LINK;
    const tree = shallow(<DisplayWork link={link.currentWork} />);

    expect(tree).toMatchSnapshot();
  });
});
describe('DisplaySocial Preview', () => {
  it('renders correctly', () => {
    let social = DEFAULT_SOCIAL;
    const tree = shallow(
      <DisplaySocial
        base="https://codepen.io"
        icon="https://cdn.jsdelivr.net/npm/simple-icons@3.0.1/icons/codepen.svg"
        username={social.codepen}
      />
    );

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no username', () => {
    const tree = shallow(
      <DisplaySocial
        base="https://codepen.io"
        icon="https://cdn.jsdelivr.net/npm/simple-icons@3.0.1/icons/codepen.svg"
        username={''}
      />
    );

    expect(tree).toMatchSnapshot();
  });
});
describe('VisitorsBadge Preview', () => {
  it('renders correctly', () => {
    let data = DEFAULT_DATA;
    let social = DEFAULT_SOCIAL;
    const tree = shallow(
      <VisitorsBadgePreview
        show={data.visitorsBadge}
        github={social.github}
        badgeOptions={{
          badgeLabel: encodeURI(data.badgeLabel),
          badgeColor: data.badgeColor,
          badgeStyle: data.badgeStyle,
        }}
      />
    );

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with show true', () => {
    let data = DEFAULT_DATA;
    let social = DEFAULT_SOCIAL;
    const tree = shallow(
      <VisitorsBadgePreview
        show={true}
        github={social.github}
        badgeOptions={{
          badgeLabel: encodeURI(data.badgeLabel),
          badgeColor: data.badgeColor,
          badgeStyle: data.badgeStyle,
        }}
      />
    );

    expect(tree).toMatchSnapshot();
  });
});
describe('GithubProfileTrophy Preview', () => {
  it('renders correctly', () => {
    let data = DEFAULT_DATA;
    let social = DEFAULT_SOCIAL;
    const tree = shallow(<GithubProfileTrophyPreview show={data.githubProfileTrophy} github={social.github} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with show true', () => {
    let data = DEFAULT_DATA;
    let social = DEFAULT_SOCIAL;
    const tree = shallow(<GithubProfileTrophyPreview show={true} github={social.github} />);

    expect(tree).toMatchSnapshot();
  });
});
describe('TwitterBadgePreview Preview', () => {
  it('renders correctly', () => {
    let data = DEFAULT_DATA;
    let social = DEFAULT_SOCIAL;
    const tree = shallow(<TwitterBadgePreview show={data.twitterBadge} twitter={social.twitter} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with show true', () => {
    let data = DEFAULT_DATA;
    let social = DEFAULT_SOCIAL;
    const tree = shallow(<TwitterBadgePreview show={true} twitter={social.twitter} />);

    expect(tree).toMatchSnapshot();
  });
});
describe('Work Preview', () => {
  it('renders correctly', () => {
    let data = DEFAULT_DATA;
    let prefix = DEFAULT_PREFIX;
    let link = DEFAULT_LINK;
    let props = { data: data, prefix: prefix, link: link };
    const tree = shallow(<WorkPreview work={props} />);

    expect(tree).toMatchSnapshot();
  });
});
describe('Social Preview', () => {
  it('renders correctly', () => {
    let social = DEFAULT_SOCIAL;
    const tree = shallow(<SocialPreview social={social} />);

    expect(tree).toMatchSnapshot();
  });
});
describe('Skills Preview', () => {
  it('renders correctly', () => {
    let skills = DUMMY_SKILLS.skills;
    const tree = shallow(<SkillsPreview skills={skills} />);

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with no skills', () => {
    let skills = {};
    const tree = shallow(<SkillsPreview skills={skills} />);

    expect(tree).toMatchSnapshot();
  });
});
describe('TopLanguages Preview', () => {
  it('renders correctly', () => {
    let data = DEFAULT_DATA;
    let social = DEFAULT_SOCIAL;
    const tree = shallow(
      <TopLanguagesPreview show={data.topLanguages} github={social.github} options={data.topLanguagesOptions} />
    );

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with show true', () => {
    let data = DEFAULT_DATA;
    let social = DEFAULT_SOCIAL;
    const tree = shallow(<TopLanguagesPreview show={true} github={social.github} options={data.topLanguagesOptions} />);

    expect(tree).toMatchSnapshot();
  });
});
describe('GitHubStats Preview', () => {
  it('renders correctly', () => {
    let data = DEFAULT_DATA;
    let social = DEFAULT_SOCIAL;
    const tree = shallow(
      <GitHubStatsPreview show={data.githubStats} github={social.github} options={data.githubStatsOptions} />
    );

    expect(tree).toMatchSnapshot();
  });
  it('renders correctly', () => {
    let data = DEFAULT_DATA;
    let social = DEFAULT_SOCIAL;
    const tree = shallow(<GitHubStatsPreview show={true} github={social.github} options={data.githubStatsOptions} />);

    expect(tree).toMatchSnapshot();
  });
});
