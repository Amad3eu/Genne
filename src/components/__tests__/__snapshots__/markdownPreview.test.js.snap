// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplaySocial Preview renders correctly 1`] = `
<a
  className="no-underline text-blue-700 m-2"
  href="https://codepen.io/dummy"
  target="blank"
>
  <img
    alt="username"
    className="w-6 h-6"
    src="https://cdn.jsdelivr.net/npm/simple-icons@3.0.1/icons/codepen.svg"
  />
</a>
`;

exports[`DisplaySocial Preview renders correctly with no username 1`] = `""`;

exports[`DisplayWork Preview renders correctly 1`] = `
<div
  className="my-2"
>
  [object Object] 
  <a
    className="no-underline text-blue-700"
    href="https://dummy.com"
    target="blank"
  >
    readme-generator
  </a>
</div>
`;

exports[`DisplayWork Preview renders correctly with no link 1`] = `
<div
  className="my-2"
>
  [object Object] 
  <b>
    readme-generator
  </b>
</div>
`;

exports[`DisplayWork Preview renders correctly with no prefix 1`] = `""`;

exports[`DisplayWork Preview renders correctly with no prefix and link 1`] = `""`;

exports[`DisplayWork Preview renders correctly with no prefix, link and project 1`] = `""`;

exports[`DisplayWork Preview renders correctly with no project 1`] = `
<div
  className="my-2"
>
  [object Object] 
  <a
    className="no-underline text-blue-700"
    href="https://dummy.com"
    target="blank"
  >
    https://dummy.com
  </a>
</div>
`;

exports[`DisplayWork Preview renders correctly with no project and link 1`] = `""`;

exports[`DisplayWork Preview renders correctly with no project and prefix 1`] = `""`;

exports[`GitHubStats Preview renders correctly 1`] = `""`;

exports[`GitHubStats Preview renders correctly 2`] = `
<div
  className="text-center mx-4 mb-4"
>
  <img
    alt=""
    src="https://github-readme-stats.vercel.app/api?username=&show_icons=true&locale=en"
  />
</div>
`;

exports[`GithubProfileTrophy Preview renders correctly 1`] = `""`;

exports[`GithubProfileTrophy Preview renders correctly with show true 1`] = `
<div
  className="text-left my-2"
>
   
  <a
    href="https://github.com/ryo-ma/github-profile-trophy"
  >
    <img
      alt=""
      src="https://github-profile-trophy.vercel.app/?username="
    />
  </a>
   
</div>
`;

exports[`Markdown Preview renders correctly 1`] = `
<div
  id="markdown-preview"
>
  <TitlePreview
    prefix="Hi 👋, I'm"
    title="dummy"
  />
  <SubTitlePreview
    subtitle="A passionate frontend developer from India"
  />
  <VisitorsBadgePreview
    badgeOptions={
      Object {
        "badgeColor": "0e75b6",
        "badgeLabel": "Profile%20views",
        "badgeStyle": "flat",
      }
    }
    github=""
    show={false}
  />
  <GithubProfileTrophyPreview
    github=""
    show={false}
  />
  <TwitterBadgePreview
    show={false}
    twitter=""
  />
  <WorkPreview
    work={
      Object {
        "data": Object {
          "ama": "",
          "badgeColor": "0e75b6",
          "badgeLabel": "Profile views",
          "badgeStyle": "flat",
          "collaborateOn": "",
          "contact": "",
          "currentLearn": "",
          "currentWork": "readme-generator",
          "devDynamicBlogs": false,
          "funFact": "",
          "githubProfileTrophy": false,
          "githubStats": false,
          "githubStatsOptions": Object {
            "bgColor": "",
            "cacheSeconds": null,
            "hideBorder": false,
            "locale": "en",
            "textColor": "",
            "theme": "",
            "titleColor": "",
          },
          "helpWith": "",
          "mediumDynamicBlogs": false,
          "rssDynamicBlogs": false,
          "subtitle": "A passionate frontend developer from India",
          "title": "dummy",
          "topLanguages": false,
          "topLanguagesOptions": Object {
            "bgColor": "",
            "cacheSeconds": null,
            "hideBorder": false,
            "locale": "en",
            "textColor": "",
            "theme": "",
            "titleColor": "",
          },
          "twitterBadge": false,
          "visitorsBadge": false,
        },
        "link": Object {
          "blog": "",
          "collaborateOn": "",
          "currentWork": "https://dummy.com",
          "helpWith": "",
          "portfolio": "",
          "resume": "",
        },
        "prefix": Object {
          "ama": "💬 Ask me about",
          "blog": "📝 I regularly write articles on",
          "collaborateOn": "👯 I’m looking to collaborate on",
          "contact": "📫 How to reach me",
          "currentLearn": "🌱 I’m currently learning",
          "currentWork": "🔭 I’m currently working on",
          "funFact": "⚡ Fun fact",
          "helpWith": "🤝 I’m looking for help with",
          "portfolio": "👨‍💻 All of my projects are available at",
          "resume": "📄 Know about my experiences",
          "title": "Hi 👋, I'm",
        },
        "skills": Object {},
        "social": Object {
          "behance": "",
          "codechef": "",
          "codeforces": "",
          "codepen": "dummy",
          "codesandbox": "",
          "dev": "",
          "discord": "",
          "dribbble": "",
          "fb": "",
          "geeks_for_geeks": "",
          "github": "",
          "hackerearth": "",
          "hackerrank": "",
          "instagram": "",
          "kaggle": "",
          "leetcode": "",
          "linkedin": "",
          "medium": "",
          "rssurl": "",
          "stackoverflow": "",
          "topcoder": "",
          "twitter": "",
          "youtube": "",
        },
        "support": Object {
          "buyMeACoffee": "",
        },
      }
    }
  />
  <SocialPreview
    social={
      Object {
        "behance": "",
        "codechef": "",
        "codeforces": "",
        "codepen": "dummy",
        "codesandbox": "",
        "dev": "",
        "discord": "",
        "dribbble": "",
        "fb": "",
        "geeks_for_geeks": "",
        "github": "",
        "hackerearth": "",
        "hackerrank": "",
        "instagram": "",
        "kaggle": "",
        "leetcode": "",
        "linkedin": "",
        "medium": "",
        "rssurl": "",
        "stackoverflow": "",
        "topcoder": "",
        "twitter": "",
        "youtube": "",
      }
    }
  />
  <SkillsPreview
    skills={Object {}}
  />
  <SupportPreview
    support={
      Object {
        "buyMeACoffee": "",
      }
    }
  />
  <div
    className="block sm:flex sm:justify-center sm:items-start"
  >
    <TopLanguagesPreview
      github=""
      options={
        Object {
          "bgColor": "",
          "cacheSeconds": null,
          "hideBorder": false,
          "locale": "en",
          "textColor": "",
          "theme": "",
          "titleColor": "",
        }
      }
      show={false}
    />
    <GitHubStatsPreview
      github=""
      options={
        Object {
          "bgColor": "",
          "cacheSeconds": null,
          "hideBorder": false,
          "locale": "en",
          "textColor": "",
          "theme": "",
          "titleColor": "",
        }
      }
      show={false}
    />
    <StreakStatsPreview
      github=""
      options={Object {}}
      show={false}
    />
  </div>
</div>
`;

exports[`SectionTitle Preview renders correctly 1`] = `
<h3
  className="w-full text-lg sm:text-xl"
>
  dummy
</h3>
`;

exports[`SectionTitle Preview renders correctly with no label 1`] = `""`;

exports[`SectionTitle Preview renders correctly with visible false 1`] = `""`;

exports[`Skills Preview renders correctly 1`] = `
<div
  className="flex flex-wrap justify-start items-center"
>
  <SectionTitle
    label="Languages and Tools:"
    visible={true}
  />
  <a
    href="https://unity.com/"
    key="unity"
    rel="noreferrer"
    target="_blank"
  >
    <img
      alt="unity"
      className="mb-4 mr-4 h-6 w-6 sm:h-10 sm:w-10"
      src="https://www.vectorlogo.zone/logos/unity3d/unity3d-icon.svg"
    />
  </a>
</div>
`;

exports[`Skills Preview renders correctly with no skills 1`] = `""`;

exports[`Social Preview renders correctly 1`] = `
<div
  className="flex justify-start items-end flex-wrap"
>
  <SectionTitle
    label="Connect with me:"
    visible={true}
  />
  <DisplaySocial
    base="https://codepen.io"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/codepen.svg"
    username="dummy"
  />
  <DisplaySocial
    base="https://dev.to"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/devto.svg"
    username=""
  />
  <DisplaySocial
    base="https://twitter.com"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/twitter.svg"
    username=""
  />
  <DisplaySocial
    base="https://linkedin.com/in"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/linked-in-alt.svg"
    username=""
  />
  <DisplaySocial
    base="https://stackoverflow.com/users"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/stack-overflow.svg"
    username=""
  />
  <DisplaySocial
    base="https://codesandbox.com"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/codesandbox.svg"
    username=""
  />
  <DisplaySocial
    base="https://kaggle.com"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/kaggle.svg"
    username=""
  />
  <DisplaySocial
    base="https://fb.com"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/facebook.svg"
    username=""
  />
  <DisplaySocial
    base="https://instagram.com"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/instagram.svg"
    username=""
  />
  <DisplaySocial
    base="https://dribbble.com"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/dribbble.svg"
    username=""
  />
  <DisplaySocial
    base="https://www.behance.net"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/behance.svg"
    username=""
  />
  <DisplaySocial
    base="https://medium.com"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/medium.svg"
    username=""
  />
  <DisplaySocial
    base="https://www.youtube.com/c"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/youtube.svg"
    username=""
  />
  <DisplaySocial
    base="https://www.codechef.com/users"
    icon="https://cdn.jsdelivr.net/npm/simple-icons@3.1.0/icons/codechef.svg"
    username=""
  />
  <DisplaySocial
    base="https://www.hackerrank.com"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/hackerrank.svg"
    username=""
  />
  <DisplaySocial
    base="https://codeforces.com/profile"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/codeforces.svg"
    username=""
  />
  <DisplaySocial
    base="https://www.leetcode.com"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/leet-code.svg"
    username=""
  />
  <DisplaySocial
    base="https://www.hackerearth.com"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/hackerearth.svg"
    username=""
  />
  <DisplaySocial
    base="https://auth.geeksforgeeks.org/user"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/geeks-for-geeks.svg"
    username=""
  />
  <DisplaySocial
    base="https://www.topcoder.com/members"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/topcoder.svg"
    username=""
  />
  <DisplaySocial
    base="https://discord.gg"
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/discord.svg"
    username=""
  />
  <DisplaySocial
    base=""
    icon="https://raw.githubusercontent.com/rahuldkjain/github-profile-readme-generator/master/src/images/icons/Social/rss.svg"
    username=""
  />
</div>
`;

exports[`SubTitle Preview renders correctly 1`] = `
<h3
  className="text-center font-medium"
>
  A passionate frontend developer from India
</h3>
`;

exports[`SubTitle Preview renders correctly with no subtitle 1`] = `""`;

exports[`Title Preview renders correctly 1`] = `
<h1
  className="text-center text-xl font-bold"
>
  Hi 👋, I'm dummy
</h1>
`;

exports[`Title Preview renders correctly with no prefix 1`] = `""`;

exports[`Title Preview renders correctly with no title 1`] = `""`;

exports[`Title Preview renders correctly with no title and prefix 1`] = `""`;

exports[`TopLanguages Preview renders correctly 1`] = `
<div
  className="text-center mx-4 mb-4"
>
    
</div>
`;

exports[`TopLanguages Preview renders correctly with show true 1`] = `
<div
  className="text-center mx-4 mb-4"
>
  <img
    alt=""
    src="https://github-readme-stats.vercel.app/api/top-langs?username=&show_icons=true&locale=en&layout=compact"
  />
</div>
`;

exports[`TwitterBadgePreview Preview renders correctly 1`] = `""`;

exports[`TwitterBadgePreview Preview renders correctly with show true 1`] = `
<div
  className="text-left my-2"
>
   
  <a
    href="https://twitter.com/"
    rel="noreferrer"
    target="_blank"
  >
    <img
      alt=""
      className="h-4 sm:h-6"
      src="https://img.shields.io/twitter/follow/?logo=twitter&style=for-the-badge"
    />
  </a>
   
</div>
`;

exports[`VisitorsBadge Preview renders correctly 1`] = `""`;

exports[`VisitorsBadge Preview renders correctly with show true 1`] = `
<div
  className="text-left my-2"
>
   
  <img
    alt=""
    className="h-4 sm:h-6"
    src="https://komarev.com/ghpvc/?username=&label=Profile%20views&color=0e75b6&style=flat"
  />
   
</div>
`;

exports[`Work Preview renders correctly 1`] = `
<Fragment>
  <DisplayWork
    link="https://dummy.com"
    prefix="🔭 I’m currently working on"
    project="readme-generator"
  />
  <DisplayWork
    link=""
    prefix="🌱 I’m currently learning"
    project=""
  />
  <DisplayWork
    link=""
    prefix="🤝 I’m looking for help with"
    project=""
  />
  <DisplayWork
    link=""
    prefix="👯 I’m looking to collaborate on"
    project=""
  />
  <DisplayWork
    link=""
    prefix="💬 Ask me about"
    project=""
  />
  <DisplayWork
    link=""
    prefix="👨‍💻 All of my projects are available at"
    project=""
  />
  <DisplayWork
    link=""
    prefix="📝 I regularly write articles on"
    project=""
  />
  <DisplayWork
    link=""
    prefix="📄 Know about my experiences"
    project=""
  />
  <DisplayWork
    link=""
    prefix="📫 How to reach me"
    project=""
  />
  <DisplayWork
    link=""
    prefix="⚡ Fun fact"
    project=""
  />
</Fragment>
`;
